timer/base_timer.go

```diff
@@ -0,0 +1,127 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package timer
+
+import (
+	"errors"
+	"time"
+)
+
+// baseTimer is the type used for all the timers that won't have children
+// It's implementation is more lightweight than nodeTimer and can be used as a standalone timer using NewTimer
+type baseTimer struct {
```

Comment:
(Optional/naming) If they don't have children, they could be named `leafTimer`

Reply:
At one point is was indeed called leafTimer but I think that it makes more sense, since this is also embedded into `nodeTimer` and used standalone-ly in `NewTimer`, to call it baseTimer. Since it's optional, ~closing~